pkg/tsdb/cloudmonitoring/time_series_query.go

```diff
@@ -132,6 +130,14 @@ func (timeSeriesQuery cloudMonitoringTimeSeriesQuery) parseResponse(queryRes *ba
 			seriesLabels["metric.name"] = d.Key
 			defaultMetricName := d.Key
 
+			// If more than 1 pointdescriptor was returned, three aggregations are returned per time series - min, mean and max.
+			// This is a because the period for the given table is less than half the duration which is used in the graph_period MQL function.
+			// See https://cloud.google.com/monitoring/mql/reference#graph_period-tabop
+			// When this is the case, we'll just ignore the min and max and use the mean value in the frame
+			if len(response.TimeSeriesDescriptor.PointDescriptors) > 1 && !strings.HasSuffix(d.Key, ".mean") {
+				continue
+			}
```

Comment:
maybe, it's better (for readability) to pre-process the `PointDescriptors` and find the one you are interested in rather than iterating over them and returning early

Reply:
Maybe, but since this is just a bug fix I'm a bit on the defensive side here - don't want to introduce other problems. Maybe save that one for later if ok?